apis/supervisor/config/v1alpha1/types_federationdomain.go.tmpl

```diff
@@ -58,6 +207,32 @@ type FederationDomainSpec struct {
 	// TLS configures how this FederationDomain is served over Transport Layer Security (TLS).
 	// +optional
 	TLS *FederationDomainTLSSpec `json:"tls,omitempty"`
+
+	// IdentityProviders is the list of identity providers available for use by this FederationDomain.
+	//
+	// An identity provider (e.g. OIDCIdentityProvider or LDAPIdentityProvider) describes how to connect to a server,
+	// how to talk in a specific protocol for authentication, and how to use the schema of that server/protocol to
+	// extract a normalized user identity. Normalized user identities include a username and a list of group names.
+	// In contrast, IdentityProviders describes how to use that normalized identity in a group of
+	// Kubernetes clusters. It can perform arbitrary transformations on that normalized identity. For example, a
+	// transformation can add a prefix to all usernames to help avoid accidental conflicts when multiple identity
+	// providers have different users with the same username (e.g. "idp1:ryan" versus "idp2:ryan").
+	// IdentityProviders can also implement arbitrary authentication rejection policies.
+	// Even though a user was able to authenticate with the identity provider, a policy can disallow the authentication
+	// to the Kubernetes clusters that belong to this FederationDomain. For example, a policy could disallow the
+	// authentication unless the user belongs to a specific group in the identity provider.
+	//
+	// For backwards compatibility with versions of Pinniped which predate support for multiple identity providers,
+	// an empty IdentityProviders list will cause the FederationDomain to use all available identity providers which
+	// exist in the same namespace, but also to reject all authentication requests when there is more than one identity
+	// provider currently defined. In this backwards compatibility mode, the name of the identity provider resource
+	// (e.g. the Name of an OIDCIdentityProvider resource) will be used as the name of the identity provider in this
+	// FederationDomain. This mode is provided to make upgrading from older versions easier. However, instead of
+	// relying on this backwards compatibility mode, please consider this mode to be deprecated and please instead
+	// explicitly list the identity provider using this IdentityProviders field.
+	//
+	// +optional
+	IdentityProviders []FederationDomainIdentityProvider `json:"identityProviders,omitempty"`
```

Comment:
Line 279 of this file under `type FederationDomainStatts struct` appears to have an error (refs OIDC Provider's secrets, when it is a property of `FederationDomainStatus`:

```
// Secrets contains information about this OIDC Provider's secrets.
// +optional
Secrets FederationDomainSecrets `json:"secrets,omitempty"`
```

Reply:
The FederationDomain is itself an OIDC provider. This might be a little confusing, but is technically correct. Potentially clarifying this is not related to this PR though.